### PR TITLE
chore: cleanup erpadapterrequest

### PR DESCRIPTION
**Pull request by eschrewe**
## Description
- added validation for currently unsupported request types
- fixed erp response url
- fixed bug in PartnerControllerTest (missing mock)
- solves #419 

## Pre-review checks

Please ensure to do as many of the following checks as possible, before asking for committer review:

- [x] DEPENDENCIES are up-to-date. [Dash license tool](https://github.com/eclipse/dash-licenses). Committers can open IP issues for restricted libs.
- [x] [Copyright and license header](https://eclipse-tractusx.github.io/docs/release/trg-7/trg-7-02) are present on all affected files
- [x] If helm chart has been changed, the chart version has been bumped to either next major, minor or patch level (compared to released chart).

### CODE REVIEW

File: backend/src/test/java/org/eclipse/tractusx/puris/backend/erpadapter/logic/service/ErpAdapterRequestServiceTest.java [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+    @InjectMocks
+    private ErpAdapterRequestService erpAdapterRequestService;
+
+    private static final String matNbrCustomer = "MNR-7307-AU340474.002";
+
+    private static final String supplierPartnerBpnl = "BPNL1234567890ZZ";
+
+    private static final String sammVersion = "2.0";
+
+
```
Comment (tom-rm-meyer-ISST) [inline backend/src/test/java/org/eclipse/tractusx/puris/backend/erpadapter/logic/service/ErpAdapterRequestServiceTest.java:56]: please remove duplicate blank. Also the ones at method end and after comments for given & then.

**Review by tom-rm-meyer-ISST — CHANGES_REQUESTED** [review]
Thanks a lot (also for fixing the `PartnerControllerTest`)! Checked the issue again: Currently there is still the need to provide tests for the mapping of responses.
Note: as this PR is concise I could also split the issue into two tickets. Please let me know what you prefer :)

**Comment by eschrewe** [comment]
After taking a closer look, I noticed that the Asset Type validation was not working correctly in the Unit tests, so I did some refactoring here. 

Regarding the issue of adding additional tests for the mapping of responses, I think it'd be best if we wait for detailed feedback from the other team, that is currently supposed to work on the erp-adapter project (Gerhard). Maybe they come up with a request to change to API a bit, or something like that...

**Comment by tom-rm-meyer-ISST** [comment]
> After taking a closer look, I noticed that the Asset Type validation was not working correctly in the Unit tests, so I did some refactoring here.
> 
> Regarding the issue of adding additional tests for the mapping of responses, I think it'd be best if we wait for detailed feedback from the other team, that is currently supposed to work on the erp-adapter project (Gerhard). Maybe they come up with a request to change to API a bit, or something like that...

One could argue for both as likely changes won't be fundamental (I expect). As proposed I've split #419 into #584 and #583. This issue solves #584 

Is this pr ready for review? :)

**Comment by eschrewe** [comment]
> > After taking a closer look, I noticed that the Asset Type validation was not working correctly in the Unit tests, so I did some refactoring here.
> > Regarding the issue of adding additional tests for the mapping of responses, I think it'd be best if we wait for detailed feedback from the other team, that is currently supposed to work on the erp-adapter project (Gerhard). Maybe they come up with a request to change to API a bit, or something like that...
> 
> One could argue for both as likely changes won't be fundamental (I expect). As proposed I've split #419 into #584 and #583. This issue solves #584
> 
> Is this pr ready for review? :)

Yes

**Review by tom-rm-meyer-ISST — APPROVED** [review]
LGTM, thanks for contributing!

**Pull request merged**